### PR TITLE
Cleanup reprojection

### PR DESCRIPTION
**Pull request by joto**
Some steps cleaning up the reprojection code. See individual commits.

### CODE REVIEW

File: src/expire-tiles.cpp [hunk 1]
```diff
@@ -122,6 +122,21 @@ int expire_tiles::normalise_tile_x_coord(int x)
...
 }
 
+/**
+ * Converts from target coordinates to tile coordinates.
+ *
+ * The zoom level for the coordinates is explicitly given in the
+ * parameter map_width.
+ */
+static void coords_to_tile(reprojection const &projection, double *tilex,
+                           double *tiley, double lon, double lat, int map_width)
```
Comment (lonvia) [inline src/expire-tiles.cpp:132]: If you make that a member of expire_tiles, you can save the projection and map_width parameter. And while you are at it, I would suggest to move output parameters to the end of the parameter list.

File: src/reprojection.cpp [hunk 2]
```diff
@@ -1,74 +1,59 @@
...
 
 #include "reprojection.hpp"
 
-/** must match expire.tiles.c */
-#define EARTH_CIRCUMFERENCE 40075016.68
-
 namespace {
 
-void latlon2merc(double *lat, double *lon)
+Coordinates lonlat2merc(Coordinates coords)
```
Comment (lonvia) [inline src/reprojection.cpp:7]: Shouldn't it rather be coord2merc now?

File: src/reprojection.hpp [hunk 3]
```diff
@@ -35,28 +45,22 @@ class reprojection
...
 
     /**
-     * Converts coordinates from target projection to tile projection (EPSG:3857)
+     * Converts coordinates from target projection to tile projection
+     * (EPSG:3857)
      *
-     * Do not confuse with coords_to_tile which does *not* calculate coordinates in the
-     * tile projection, but tile coordinates.
+     * Do not confuse with coords_to_tile which does *not* calculate
+     * coordinates in the tile projection, but tile coordinates.
```
Comment (lonvia) [inline src/reprojection.hpp:55]: Remove this comment now that coords_to_tile has gone away?

File: src/reprojection.cpp (hunk 2 above)
Comment (joto) [reply src/reprojection.cpp:7]: It still converts coordinates from lon/lat format to Mercator. The `Coordinates` just holds two doubles and can be used by many different coordinate systems.

File: src/expire-tiles.cpp (hunk 1 above)
Comment (joto) [reply src/expire-tiles.cpp]: I added a commit with this.

File: src/reprojection.hpp (hunk 3 above)
Comment (joto) [reply src/reprojection.hpp]: Adressed in one of the commits.

**Pull request merged**